pyproj/_datadir.pyx

```diff
@@ -19,7 +19,8 @@ _USE_GLOBAL_CONTEXT = strtobool(os.environ.get("PYPROJ_GLOBAL_CONTEXT", "OFF"))
 # static user data directory to prevent core dumping
 # see: https://github.com/pyproj4/pyproj/issues/678
 cdef const char* _USER_DATA_DIR = proj_context_get_user_writable_directory(NULL, False)
-
+# Store the message from any internal PROJ errors
+cdef str _INTERNAL_PROJ_ERROR = None
```

Comment:
I'm not sure how this behaves across threads compared to the previous implementation of using class attributes to store the information. Do I need to grab a context here when using this?

Reply:
It looks like the type is string, so I imagine that it still uses the Python gil. Due to that, I believe that the behavior here should be consistent with the previous behavior.

Reply:
Another thing to think about is whether you want `get_proj_error()` and `clear_proj_error()` to leak into the public modules or whether you want those kept as private functions.

Reply:
I think that keeping them private is a good idea.